File: LCObjective-C/LCObjective-C/Problems/TwoSum/TwoSum.m
//
//  TwoSum.m
//  LCObjective-C
//
//  Created by Sherzod Akhmedov on 23/10/21.
//

#import <Foundation/Foundation.h>
#import "TwoSum.h"

@implementation TwoSum

+ (NSArray *)twoSum:(NSArray *)nums target:(NSInteger)t
{
    NSInteger totalCount = [nums count];
    
    if (totalCount == 0 || totalCount < 2) {
        return [[NSArray alloc] init];
    }
    
    for (NSInteger xIndex = 0; xIndex < totalCount - 1; xIndex++) {
        for (NSInteger yIndex = xIndex + 1; yIndex < totalCount; yIndex++) {
            NSInteger x = [[nums objectAtIndex:xIndex] integerValue];
            NSInteger y = [[nums objectAtIndex:yIndex] integerValue];
            if (x + y == t) {
                return [[NSArray alloc] initWithObjects:[NSNumber numberWithInteger:xIndex], [NSNumber numberWithInteger:yIndex], nil];
            }
        }
    }
    
    return [[NSArray alloc] init];
}

@end

File: LCSwift/LCSwift/Problems/TwoSum/TwoSum.swift
//
//  TwoSum.swift
//  LCSwift
//
//  Created by Sherzod Akhmedov on 23/10/21.
//

import Foundation

class TwoSum {
    static func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        if nums.isEmpty {
            return []
        }
        let totalCount = nums.count
        for xIndex in 0..<totalCount - 1 {
            for yIndex in (xIndex + 1)..<totalCount {
                if nums[xIndex] + nums[yIndex] == target {
                    return [xIndex, yIndex]
                }
            }
        }
        return []
    }
}
